public/js/contribute/pad/save.js

```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
     datum.has_content = datum.txt?.trim()?.length > 0;
     // items.push(datum);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
-      // text += `${datum.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
+    let innerText = '';
+    if (datum.has_content && datum.txt) innerText += datum.txt;
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${datum.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
     }
-    if (datum.has_content) text += datum.txt;
-
-    return { item: datum, text }
 
+    return { item: datum, text };
   } else if (datum.type === 'embed') {
     datum.html = (
       sel.select('.media-embed').node() || sel.select('.meta-embed').node()
     ).innerHTML;
     datum.has_content = datum.html?.trim()?.length > 0;
     // items.push(datum);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
-      // text += `${datum.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
-    }
+    let innerText = '';
     if (datum.has_content) {
-      text += (sel.select('.media-embed').node() ||
+      const newText = (sel.select('.media-embed').node() ||
         sel.select('.meta-embed').node())[
         'outerText' || 'textContent' || 'innerText'
       ]; // HERE WE DO NOT WANT THE html TAGS IN THE fullTxt
+      if (newText) {
+        innerText += newText;
+      }
     }
-
-    return { item: datum, text }
-
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${datum.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
+    }
+    return { item: datum, text };
   } else if (['checklist', 'radiolist'].includes(datum.type)) {
     datum.has_content =
       datum.options.filter((b) => b.name?.length && b.checked).length > 0;
     const clone = JSON.parse(JSON.stringify(datum));
     clone.options = clone.options.filter((b) => b.name?.length);
     // items.push(clone);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
-      // text += `${clone.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
+    let innerText = '';
+    if (datum.has_content)
+      innerText += clone.options
+        .filter((b) => b.name?.length && b.checked)
+        .map((b) => b.name)
+        .join('\n');
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${clone.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
```

Comment:
again, this seems quite a lengthy rewrite to (if I get it correctly) simply check for 0 length strings. what am I missing?

Reply:
innerText is needed to delay the update. innerText can be completely made of white space